modules/gh/gh.go

```diff
@@ -0,0 +1,291 @@
+//go:build gh
+// +build gh
+
+package gh
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	goos "os"
+	"strings"
+
+	"github.com/risor-io/risor/internal/arg"
+	"github.com/risor-io/risor/object"
+	"github.com/risor-io/risor/os"
+)
+
+func IsDebug(ctx context.Context, args ...object.Object) object.Object {
+	if err := arg.Require("gh.is_debug", 0, args); err != nil {
+		return err
+	}
+	env, ok := os.GetDefaultOS(ctx).LookupEnv("RUNNER_DEBUG")
+	if !ok {
+		return object.False
+	}
+	if env != "1" {
+		return object.False
+	}
+	return object.True
+}
+
+func printableValue(obj object.Object) any {
+	if iface := obj.Interface(); iface != nil {
+		return iface
+	}
+	switch obj := obj.(type) {
+	case fmt.Stringer:
+		return obj.String()
+	default:
+		return obj.Inspect()
+	}
+}
+
+var workflowCommandReplacer = strings.NewReplacer(
+	// Trick to get newlines included
+	// https://github.com/actions/toolkit/issues/193#issuecomment-605394935
+	"\r\n", "%0A",
+	"\r", "%0A",
+	"\n", "%0A",
+	"::", "%3A%3A",
+)
+
+func sanitizeCommandValue(value string) string {
+	return workflowCommandReplacer.Replace(strings.TrimSuffix(value, "\n"))
+}
+
+func asCommandProps(obj *object.Map) map[string]any {
+	props := make(map[string]any)
+	for _, key := range obj.StringKeys() {
+		cmdKey, ok := commandPropKey(key)
+		if !ok {
+			continue
+		}
+		props[cmdKey] = printableValue(obj.Get(key))
+	}
+	return props
+}
+
+func commandPropKey(key string) (string, bool) {
+	switch key {
+	case "title", "file", "line":
+		return key, true
+	case "column":
+		return "col", true
+	case "end_line":
+		return "endLine", true
+	case "end_column":
+		return "endColumn", true
+	default:
+		return "", false
+	}
+}
+
+func stringifyCommandProps(props map[string]any) string {
+	var sb strings.Builder
+	for key, value := range props {
+		if sb.Len() == 0 {
+			sb.WriteByte(' ')
+		} else {
+			sb.WriteByte(',')
+		}
+		sb.WriteString(key)
+		sb.WriteByte('=')
+		sb.WriteString(sanitizeCommandValue(fmt.Sprint(value)))
+	}
+	return sb.String()
+}
+
+func runWorkflowCommand(w io.Writer, cmd string, value any, props map[string]any) object.Object {
+	if _, err := fmt.Fprintf(w, "::%s%s::%s\n",
+		cmd,
+		stringifyCommandProps(props),
+		sanitizeCommandValue(fmt.Sprint(value))); err != nil {
+		return object.Errorf("io error: %v", err)
+	}
+	return object.Nil
+}
+
+func appendWorkflowFile(path string, message string) object.Object {
+	// Using the Go "os" package instead of the Risor "os" package
+	// because the Risor application might use S3 as storage, but
+	// the GitHub Actions special files are still on the real OS' file system.
+	file, err := goos.OpenFile(path, goos.O_APPEND|goos.O_WRONLY|goos.O_CREATE, 0644)
+	if err != nil {
+		return object.Errorf("io error: %v", err)
+	}
```

Comment:
While it's true that Risor _could_ be configured to use s3 storage for the entire filesystem, I don't think that would be likely in a situation involving Github actions. Let's still use Risor's GetDefaultOS mechanism here. Direct OS access circumvents using Risor in sandboxing like situations.

Reply:
Alright. I've added a `OS.Append()` function to the OS interface so I could completely rely on the Risor OS